src/lib/date-utils.ts

```
@@ -305,3 +282,158 @@ export function getIntervalTypeTitle(intervalType: IntervalType): string {
+ * Formats interval date for URL path segments
+ * @param interval - The time interval
+ * @returns Formatted date string or array based on interval type
+ */
+export function formatIntervalForPath(interval: TimeInterval): string[] {
+  switch (interval.intervalType) {
+    case "day":
+      return [toDateString(interval.intervalStart)];
+    case "week":
+      return [toDateString(interval.intervalStart)];
+    case "month":
+      const date = interval.intervalStart;
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)